added tools

sibylobodoekwe opened the pull request:
Added the Base64 page in src/pages.
Added the Cloud Code page in src/pages.
Added the JSON Validator page in src/pages.

sibylobodoekwe committed:
added tools

risenW commented:
> Added the Base64 page in src/pages. Added the Cloud Code page in src/pages. Added the JSON Validator page in src/pages.

Hey @sibylobodoekwe Great first contribution. I need some clarification on these new features you're proposing. 
Ideally, this should be discussed first in the issue you create before working on a PR, but, nvm, that's my fault for not adding a guide. 

-  Base64 encoder/decoder: What does this do? and how useful is it? I tried to play with the utility and don't know what it does. 
- Cloud Code: Same as above. What does this do?
- JSON Validator: Sadly there is a JSON viewer already which does validation

Let me know what the other two really do, and we can see how to modify them together.

sibylobodoekwe commented:
hi @risenW, thank you so much for the feedback.. here's a small description of base64 and cloudcode

-  **Base64 encoder/decoder** is an encoding scheme for converting binary data into a text format that can be safely transmitted over text-based protocols. Represents binary data in an ASCII string format.
- **Cloud Code** is a server-side code or scripts running on cloud computing platforms for scalable and managed execution of server-side logic. Allows developers to build/deploy server-side logic; handling API requests, data processing, background tasks, etc in a scalable and managed environment provided by the cloud platform.- 

i see they also need add placeholders descriptions added to their _index.tsx_ as well. 
They are open for improvements/contributions as i only built simplified versions of them as i expected only 1 or 2 to be approved so yea i agree the validator is not a great contender. 
Happy to work on modifications, thank you.

_(i've added a concise description of all 3 to the issues comment as well)_

risenW commented:
> * **Cloud Code** is a server-side code or scripts running on cloud computing platforms for scalable and managed execution of server-side logic. Allows developers to build/deploy server-side logic; handling API requests, data processing, background tasks, etc in a scalable and managed environment provided by the cloud platform.-


Okay, a couple of things for you to do:

1. Take out the JSON validator, as it already exists
2. Cloud code is not useful for this app, so take it out as well.
3. You can keep the base64 encoder, but adjust the image so it's the same size
![Screenshot 2023-07-17 at 11 57 55 am](https://github.com/risenW/random-abcs/assets/29900845/83265907-efe9-4d59-be06-169a655df8a3)

sibylobodoekwe commented:
alright great,  i'd get on it right now.

sibylobodoekwe committed:
Update base64.tsx, Update index.tsx, and delete unused files

sibylobodoekwe commented:
Here are the changes i made;
- I modified and improved on the base64 features on src/pages
- I adjusted image size for base64 on src/index
- deleted unapproved tools (cloudcode and jsonvalidator) features and images from  random abc folder

Thank for your feedback and the opportunity, happy to make any improvements or modifications needed.

risenW commented:
@sibylobodoekwe 
I went through your PR again. Here are some things to improve:

1. Use AntD components for things like Button, Form Input etc. See examples on other pages
2. Update page style and format. Make it similar to the other pages. That is options can be on the left or right and input boxes can be on either side.  Something like the following:
<img width="1290" alt="Screenshot 2023-07-18 at 5 51 02 pm" src="https://github.com/risenW/random-abcs/assets/29900845/2f589b36-7f65-4be2-9fc5-7d3f1cacc1b9">

Always refer to the existing page/style, and try to match them. You can copy an existing style and just add your new features

3. Lastly fix merge conflict with yarn file

sibylobodoekwe commented:
Oh thanks, just saw this, would get on it and update.

sibylobodoekwe committed:
Modified base64 style layout with Antd

sibylobodoekwe committed:
modified:   src/pages/base64.tsx
        modified:   src/pages/index.tsx
        modified:   yarn.lock

sibylobodoekwe committed:
Yarn.lock update

sibylobodoekwe committed:
Merge branch 'main' into main
